### PR TITLE
feat: [CDS-71744]: studio changes for retry step group as FS

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Studio changes for retry step group as failure strategy for step group

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/52395b5e-f850-431a-9c37-1b94fc7b1a7a

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by vikrant-harness** [comment]
retrigger titlecheck

File: src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/FailureStrategyPanel/FailureStrategyPanel.tsx [hunk 1]
```diff
@@ -38,10 +44,17 @@ export interface FailureStrategyPanelProps {
...
 }
 
 export function FailureStrategyPanel(props: FailureStrategyPanelProps): React.ReactElement {
-  const { mode, path = 'failureStrategies', isReadonly, stageType = StageType.DEPLOY } = props
+  const {
+    mode,
+    path = 'failureStrategies',
+    isReadonly,
+    stageType = StageType.DEPLOY,
+    isStepInsideStepGroup = 'false'
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/FailureStrategyPanel/FailureStrategyPanel.tsx:56]: We should have boolean value here as default value
isStepInsideStepGroup = false

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/FailureStrategyPanel/FailureStrategyPanel.tsx (hunk 1 above)
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/FailureStrategyPanel/FailureStrategyPanel.tsx:56]:
```suggestion
    isStepInsideStepGroup = false
```
This is of boolean type

File: src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/FailureStrategyPanel/utils.ts [hunk 2]
```diff
@@ -85,3 +86,5 @@ export function handleChangeInStrategies({
 export function getTabIntent(i: number, selectedStrategyNum: number): Intent {
   return i === selectedStrategyNum ? Intent.PRIMARY : Intent.NONE
 }
+
+export const EXCLUDE_STEP_INSIDE_STEP_GROUP_FAILURE_STRAT_LIST = [Strategy.RetryStepGroup]
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineSteps/AdvancedSteps/FailureStrategyPanel/utils.ts:90]:
```suggestion
export const EXCLUDE_STEP_INSIDE_STEP_GROUP_FAILURE_STRATEGY_LIST = [Strategy.RetryStepGroup]
```
We can keep the full name, as the name itself is long now

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**